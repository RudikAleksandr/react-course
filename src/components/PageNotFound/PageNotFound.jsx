import React from 'react';
import { Link } from 'react-router-dom';
import cn from './PageNotFound.module.css';

const PageNotFound = () => (
  <div className={`justify-content-center align-items-center ${cn.pageNotFound}`}>
    <h1 className={cn.notFoundLabel}>Page Not Found</h1>
    <span className={cn.errorMessage}>404</span>
    <Link className={`${cn.homeLink} btn btn-secondary`} to="/">Go back to home</Link>
  </div>
);
export default PageNotFound;
