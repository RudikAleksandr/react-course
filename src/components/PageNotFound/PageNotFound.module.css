
.pageNotFound {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background-color: #232323;
  height: 92.5vh;
  width: 100%;
}

.notFoundLabel {
  color: white;
  font-weight: 300;
}

.errorMessage {
  display: block;
  color: white;
  font-size: 7rem;
  margin-bottom: 20px;
}

.homeLink {

}