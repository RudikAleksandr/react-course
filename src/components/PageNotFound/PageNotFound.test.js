
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router';
import PageNotFound from './PageNotFound';

const component = (
  <MemoryRouter>
    <Route>
      <PageNotFound />
    </Route>
  </MemoryRouter>
);

describe('PageNotFound component', () => {
  test('render PageNotFound snapshot', () => {
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });
});