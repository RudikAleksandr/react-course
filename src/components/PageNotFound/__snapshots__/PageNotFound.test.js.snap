// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component render PageNotFound snapshot 1`] = `
<DocumentFragment>
  <div
    class="justify-content-center align-items-center pageNotFound"
  >
    <h1
      class="notFoundLabel"
    >
      Page Not Found
    </h1>
    <span
      class="errorMessage"
    >
      404
    </span>
    <a
      class="homeLink btn btn-secondary"
      href="/"
    >
      Go back to home
    </a>
  </div>
</DocumentFragment>
`;
